components/ws-manager-bridge/src/bridge-controller.ts

```diff
@@ -86,9 +86,11 @@ export class BridgeController {
         const bridge = this.bridgeFactory() as WorkspaceManagerBridge;
         const clientProvider = async () => {
             const grpcOptions = {
-                "grpc.keepalive_timeout_ms": 1500,
-                "grpc.keepalive_time_ms": 1000,
+                "grpc.keepalive_timeout_ms": 20000,
+                "grpc.http2.min_time_between_pings_ms": 1000,
```

Comment:
These two settings are too aggresive. Without changing the values I get the next error
```console
{"@type":"type.googleapis.com/google.devtools.clouderrorreporting.v1beta1.ReportedErrorEvent","serviceContext":{"service":"ws-manager-bridge","version":"aledbf-grpcjs.19"},"stack_trace":"Error: 8 RESOURCE_EXHAUSTED: Bandwidth exhausted\n    at Object.callErrorFromStatus (/app/node_modules/@grpc/grpc-js/build/src/call.js:31:26)\n    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client.js:327:49)\n    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:299:181)\n    at /app/node_modules/@grpc/grpc-js/build/src/call-stream.js:145:78\n    at processTicksAndRejections (internal/process/task_queues.js:79:11)","component":"ws-manager-bridge","severity":"ERROR","time":"2021-04-14T23:53:04.248Z","environment":"devstaging","region":"europe-west1","message":"wsman subscription error","error":"Error: 8 RESOURCE_EXHAUSTED: Bandwidth exhausted\n    at Object.callErrorFromStatus (/app/node_modules/@grpc/grpc-js/build/src/call.js:31:26)\n    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client.js:327:49)\n    at Object.onReceiveStatus (/app/node_modules/@grpc/grpc-js/build/src/client-interceptors.js:299:181)\n    at /app/node_modules/@grpc/grpc-js/build/src/call-stream.js:145:78\n    at processTicksAndRejections (internal/process/task_queues.js:79:11)","payload":{"name":"dev","url":"ws-manager:8080"}}
``` 

Is there a particular reason for those values? @csweichel @akosyakov

Reply:
We originally choose those in a time where bridge would open too many connections and swamp ws-manager. I reckon choosing more conservative values should be fine, especially if we keep tabs on ws-manager's memory consumption.